Many typo fixes (#8299)
doctrine/doctrine2: docs/en/reference/second-level-cache.rst

``Doctrine\ORM\Cache\Region\DefaultRegion`` It's the default implementation.
 A simplest cache region compatible with all doctrine-cache drivers but does not support locking.

> ``Doctrine\ORM\Cache\Region`` and ``Doctrine\ORM\Cache\ConcurrentRegion`` define contracts that should be implemented by a cache provider.

Reviewer: I think you should change line 81 too if we are going that way, also, there should be linebreaks here.

Please comply with the sf docs standards (I don't know if we follow them, but this particular rule seems appropriate):

> [Each line should break approximately after the first word that crosses the 72nd character (so most lines end up being 72-78 characters);](http://symfony.com/doc/current/contributing/documentation/standards.html)

It allows you to provide your own cache implementation that might take advantage of specific cache driver.

> If you want to support locking for ``READ_WRITE`` strategies you should implement ``ConcurrentRegion`` (``CacheRegion`` otherwise).

Reviewer: Why change this? the second part is not a precision, it's the counterpart of the first one, and has a similar importance.

Author: I reverted this change

If you want to use an ``READ_WRITE`` cache, you should consider providing your own cache region.

> ``Doctrine\ORM\Cache\ConcurrentRegion`` defines contract for concurrently managed data region.

Reviewer: ```suggestion
``Doctrine\ORM\Cache\ConcurrentRegion`` defines a contract for concurrently managed data region.
```

It allows you to provide a specific implementation of the following components :

* ``QueryCache`` stores and retrieves query cache results.
* ``CachedEntityPersister`` stores and retrieves entity results.
* ``CachedCollectionPersister`` stores and retrieves query results.

> * ``EntityHydrator``  transforms an entity into a cache entry and caches entry into entities

Reviewer 2: ```suggestion
* ``EntityHydrator``  transforms an entity into a cache entry and caches entries into entities
```
Because the sentence begins with "entity" (singular) and "entry" (singular), but ends with "entry" (singular) and "entities" (plural).

> * ``CollectionHydrator`` transforms a collection into a caches entry and cache entry into collection

Reviewer 2: ```suggestion
* ``CollectionHydrator`` transforms a collection into a cache entry and a cache entry into a collection
```
"a cache" as in "one cache".

> * ``QueryCache`` stores and retrieves query cache results.
> * ``CachedEntityPersister`` stores and retrieves entity results.
> * ``CachedCollectionPersister`` stores and retrieves query results.
> * ``EntityHydrator``  transforms an entity into a cache entry and caches entry into entities
> * ``CollectionHydrator`` transforms a collection into a caches entry and cache entry into collection

Reviewer: This could be reverted and converted to a description list (see http://html5doctor.com/the-dl-element/)

```suggestion
``QueryCache``
    stores and retrieves query cache results.
``CachedEntityPersister``
    stores and retrieves entity results.
``CachedCollectionPersister``
    stores and retrieves query results.
``EntityHydrator``
    transforms an entity into a cache entry and caches entry into entities
``CollectionHydrator``
    transforms a collection into a caches entry and cache entry into collection
```

It allows you to provide a specific implementation of the following components :

``QueryCache``
    stores and retrieves query cache results.
``CachedEntityPersister``
    stores and retrieves entity results.
``CachedCollectionPersister``
    stores and retrieves query results.
``EntityHydrator``

>     transforms an entity into a cache entry and caches entry into entities

Reviewer: ```suggestion
    transforms entities into a cache entries and cache entries into entities
```

``CollectionHydrator``

>     transforms a collection into a caches entry and cache entry into collection

Reviewer: ```suggestion
    transforms collections into cache entries and cache entries into collections
```

    $regionConfig =  $cacheConfig->getRegionsConfiguration();

    // Cache Region lifetime
    $regionConfig->setLifetime('my_entity_region', 3600);   // Time to live for a specific region (in seconds)

>     $regionConfig->setDefaultLifetime(7200);                // Default time to live  (in seconds)

Reviewer 2: ```suggestion
    $regionConfig->setDefaultLifetime(7200);                // Default time to live (in seconds)
```

    // Get the total number of cached entries *not* found in all regions.
    $logger->getMissCount();

> If you want to get more information you should implement ``\Doctrine\ORM\Cache\Logging\CacheLogger`` and collect all information you want.

Reviewer: Please use linebreaks

    // Get the total number of cached entries *not* found in all regions.
    $logger->getMissCount();

If you want to get more information you should implement
``\Doctrine\ORM\Cache\Logging\CacheLogger`` and collect

> all information you want.

Reviewer: ```suggestion
all the information you want.
```

-----------------------
* Entity cache configuration allows you to define the caching strategy and region for an entity.

>   * ``usage`` specifies the caching strategy: ``READ_ONLY``, ``NONSTRICT_READ_WRITE``, ``READ_WRITE``. See :ref:`reference-second-level-cache-mode`.

Reviewer: linebreaks

--------------------------------

As well as ``Query Cache`` all persister queries store only identifier values for an individual query.

> All persister use a single timestamp cache region to keep track of the last update for each persister,

Reviewer: ```suggestion
All persisters use a single timestamp cache region to keep track of the last update for each persister,
```

~~~~~~~~~

Count queries generated by ``Doctrine\ORM\Tools\Pagination\Paginator`` are not cached by second-level cache.

> Although entities and query result are cached,count queries will hit the

Reviewer: ```suggestion
Although entities and query result are cached, count queries will hit the
```
